### ISSUE TITLE
oxlint/lsp: Type aware does not work properly with a project in a sub-path

### ISSUE THREAD

**Issue by DavidZidar** [oxc-project/oxc#21551]
### Confirm that this is a Bug for the editor extension, not the CLI

- [x] This bug does not happen in the Oxlint/Oxfmt CLI
- [x] After restarting the editor, the bug still appears

### Versions used

```
VS Code extension: v1.52.0
oxlint: v1.60.0
oxfmt: v0.45.0
VS Code version: v1.116.0
Operating System and Version: Windows and WSL2
```

### What happened?

* When a project is in a nested folder it is not possible to enable type aware linting because the extension says that it can only be enabled in the root config even if there only is one config.
* Adding a vestigial root config to enable the setting is not a good option because then CLI linting will have to look further up the tree to find the setting for no reason.
* Furthermore the extension is unable to find the tsgolint executable (with or without a root config), see oxc-project/oxc-vscode#73, but it's even worse than that issue says because setting `oxc.path.tsgolint` to the `tsgolint.js` path does not work on Windows requiring an explicit path to a platform specific binary which makes the VS Code config not work when jumping between platforms.

### Reproduction repo

Try to use type-aware linting in a repo where the project is in a sub folder.

### Output channel log

<details>
<summary>Click to expand</summary>

```
WARN oxlint::lsp::server_linter: Failed to initialize type-aware linting: Failed to find tsgolint executable. You may need to add the `oxlint-tsgolint` package to your project?
```

</details>

### More context

Instead of trying to resolve everything from the repository root, perhaps add a working directory setting so that users such as me can specify an optional root for Oxc to work from. Having projects in sub folders is super common, so I feel that this should be resolved.

**Comment by DavidZidar** [comment]
As a workaround I've been forced to install oxlint-tsgolint globally which makes VS Code potentially use a different version of the package than CLI linting.
```shell
npm install -g oxlint-tsgolint
```

**Comment by parched** [comment]
@Sysix @connorshea @ColemanDunn I think it's fairly straight forward to make `oxc.path.tsgolint` work in a cross-platform project, we just need consensus on the desired behavior. Here's 2 possible options:
1. Add `oxc.path.tsgolint.windows` and `oxc.path.tsgolint.linux` etc., which take priority over `oxc.path.tsgolint`.
2. Check if `oxc.path.tsgolint` is a valid path, if not, try with `.cmd`, then `.exe` extension on windows and try without extension on non-windows.

I prefer option 2 and can implement it if needed.
